src/hooks/useForm/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useMemo } from 'react';
+
+// Interfaces START
+
+export interface Validator {
+  (value: string): string;
+}
+
+interface FormInput {
+  value: string;
+  validators: Validator[];
+}
+
+interface ParameterValueFunction<T> {
+  (formInput: FormInput, key: string): T;
+}
+
+interface Dictionary<T> {
+  [key: string]: T;
+}
+
+// Interfaces END
+
+// Hook START
+const useForm = (fieldsSettings: Dictionary<FormInput>) => {
+  const convertArrayToObject = (
+    dataSource: Dictionary<FormInput>,
+    createValue: ParameterValueFunction<any>
+  ) =>
+    Object.entries(dataSource).reduce(
+      (object: Dictionary<any>, item: [string, FormInput]) => {
+        let [key, value] = item;
+        object[key] = createValue(value, key);
+        return object;
+      },
+      {}
+    );
+
+  const calculateErrorsForValue = (
+    value: string,
+    validators: Validator[]
+  ): string[] => {
+    let errors: string[] = [];
+    for (let validator of validators) {
+      let errorMessage: string = validator(value);
+      if (errorMessage) {
+        errors.push(errorMessage);
+      }
+    }
+    return errors;
+  };
+
+  const getValidatorsByKey = (key: string) => {
+    return fieldsSettings[key].validators || [];
+  };
+
+  const [values, setValues] = useState<Dictionary<string>>(
+    convertArrayToObject(fieldsSettings, (formInput) => formInput.value || '')
```

Comment:
So, we have a function that is called convertArrayToObject. From this name, i assumed that we should pass an array as an argument, but checking the types of fieldSettings i see that it is already an object `fieldSettings: Dictionary<FormInput>`.
Digging a little bit more, it turns out that you need an object to be passed in to this function in order to make Object.entries work as i think it is expected to work. (notice that `Object.entries['stefano', 'carlos']` will not throw an error but [ ["0", "stefano"], ["1", "carlos"]  ] which i think its not what we want here).
I think we should review those data structures (or re-naming some variables maybe) to make some things a little less confusing

Reply:
I want to leave this functionality like this
The intention of doing so is to decouple the functionality and make reuse easier in the future

Reply:
Im not talking about the functionality here, my problem is that we are calling convertArrayToObject but NO array is being received in this function.

Reply:
this will be named as **generateObject**

do you agree? :)